### PR TITLE
Add RememberBestWeights handler: restores best weights after training

### PR DESCRIPTION
**Pull request by dnouri**

### CODE REVIEW

File: nolearn/lasagne/handlers.py [hunk 1]
```diff
@@ -95,6 +95,51 @@ def __call__(self, nn, train_history):
...
+                ))
+
+
+class RememberBestWeights:
+    def __init__(self, loss='valid_loss', score=None, verbose=1):
+        self.loss = loss
+        self.score = score
+        self.verbose = 1
+        self.best_weights = None
+        self.best_weights_loss = None
```
Comment (cancan101) [inline nolearn/lasagne/handlers.py]: rather than calling this a loss in both the case of a score and a loss, perhaps negate it? That way comparison would also be the same.

File: nolearn/lasagne/handlers.py [hunk 2]
```diff
@@ -95,6 +95,51 @@ def __call__(self, nn, train_history):
...
+        key = self.score if self.score is not None else self.loss
+
+        if self.best_weights_loss is None:
+            self.best_weights = nn.get_all_params_values()
+            self.best_weights_loss = train_history[-1][key]
+            self.best_weights_epoch = train_history[-1]['epoch']
+            return
+
+        if self.score is not None:
+            if train_history[-1][key] > self.best_weights_loss:
```
Comment (cancan101) [inline nolearn/lasagne/handlers.py]: then you would negate `train_history[-1][key]` vs `best_weights_loss` and all of the rest of the code would be the same.
